files/scripts/syncd.sh

```diff
@@ -136,6 +136,40 @@ wait() {
     /usr/bin/${SERVICE}.sh wait $DEV
 }
 
+collect_mst() {
+    debug "Collecting MST dump before syncd restart"
+    IRGDBSERVER=/usr/local/bin/ir-gdbserver
+    TMPDIR=/tmp/mlnxdump
+    MSTDIR=$TMPDIR/mstdump
+    DUMPDIR=/var/dump/mstdump
+    IRGDBDIR=$TMPDIR/ir-gdb
+    mkdir -p $MSTDIR
+    mkdir -p $IRGDBDIR
+    if ! mst status -v > $MSTDIR/mststatus; then
+        debug "mst status command returned error"
+    else
+        local mst_dump_filename="$MSTDIR/mstdump"
+        local max_dump_count="3"
+        for i in $(seq 1 $max_dump_count); do
+            if ! ${CMD_PREFIX}/usr/bin/mstdump /dev/mst/mt*pci_cr0 > "${mst_dump_filename}${i}"; then
+                debug "mstdump failed"
+                break
+            fi
+        done
+        if [ -f $IRGDBSERVER ]; then
+            debug "collecting ir-gdbserver dump"
+            $IRGDBSERVER -d /dev/mst/mt*_pci_cr0 -c $IRGDBDIR/core
+        fi
+    fi
+    mkdir -p $DUMPDIR
+    TARFILE=mstdump_`date +%Y%m%d_%H%M%S`.tar
+    tar -C $TMPDIR -cf $DUMPDIR/$TARFILE  .
+    gzip -f $DUMPDIR/$TARFILE
```

Comment:
should you print which file was created for debug purpose?

Reply:
Done